File: backend/src/main/java/org/providcy/acceptance/model/Invoice.java
package org.providcy.acceptance.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.providcy.acceptance.model.type.InvoiceState;

import javax.persistence.*;
import java.time.Instant;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "invoices")
public class Invoice {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String invoiceNumber;
    private Instant createdAt;
    private String wagonNumber;
    private Instant unloadingStart;
    private boolean unloadingError = false;
    private int weightBefore = 0;
    private int weightAfter = 0;
    private int weightCargo = 0;
    private int weightAccepted = 0;

    private String unloadingRecord;

    @Enumerated(EnumType.STRING)
    private InvoiceState state = InvoiceState.NONE;
}

File: frontend/projects/cabinet/src/app/module/invoice/model/invoice.ts
import * as moment from "moment";
import {Moment} from "moment";
import {IPager} from "../../../../../../tools/src/lib/module/common/model/pager";

export interface IInvoiceList {
    list: IInvoice[]
    pager: IPager;
    status: number
}

export interface IInvoice {
    id: number;
    invoiceNumber: string;
    createdAt: Moment | null;
    wagonNumber: string;
    unloadingStart: Moment | null;
    unloadingEnd: Moment | null;
    unloadingError: number;
    weightBefore: number;
    weightAfter: number;
    weightCargo: number;
    weightAccepted: number;
    unloadingRecord: string;
    state: string;
}

export class Invoice implements IInvoice {
    constructor(
        public id: number = 0,
        public invoiceNumber: string = '',
        public createdAt: Moment | null = null,
        public wagonNumber: string = '',
        public unloadingStart: Moment | null = null,
        public unloadingEnd: Moment | null = null,
        public unloadingError: number = 0,
        public weightBefore: number = 0,
        public weightAfter: number = 0,
        public weightCargo: number = 0,
        public weightAccepted: number = 0,
        public unloadingRecord: string = '',
        public state: string = ''
    ) {
    }

    _init(data: IInvoice | null): IInvoice {
        for (const key in data) {
            if (key === 'createdAt' || key === 'unloadingStart' || key === 'unloadingEnd') {
                if (data[key]) {
                    this[key] = moment(data[key]);
                }
            } else {
                // @ts-ignore
                this[key] = data[key];
            }
        }

        return this;
    }
}
